### DISCUSSION TITLE
xmake是否需要luajit?

### DISCUSSION THREAD

**Discussion by masx200** [xmake-io/xmake#3740] [Q&A] [answered]
xmake是否需要luajit?

**Comment by waruqi** [comment, marked as answer]
不需要，原本几年前就是用的 luajit，但作者不怎么维护，各种稳定性问题，并且跨平台性不好，不支持 riscv, loongarch 等新 arch。。

所以，近几年才切到 lua 的，而 xmake 的性能瓶颈在编译器和编译调度上，并不是计算密集型的场景，xmake 本身那点 lua 逻辑不会有太大影响。

另外切 lua 前，对比过，编译性能，两者几乎没差异，而内存占用，lua 5.4 反而更少些，另外最后生成的二进制，lua 比 luajit 少了几百 k，更加小。所以可维护性，二进制体积，内存，跨平台性都是 lua 占优。

如果你非要用 luajit，xmake 本身还是支持的，自己源码编译切换到 luajit 就行了，

```
./configure --runtime=luajit
make
```
